### PR TITLE
CalDav Setup page

### PR DESCRIPTION
**Pull request by leog**
## What does this PR do?
Moving CalDav integration setup into its own setup page as we have for Apple Calendar and Zapier.
Mouse_Highlight_Overlay
Also added a "Cancel" button to both Apple and CalDav to go back to the previous page
Included localization to the Apple Calendar setup page as it was missing

**Environment**: Staging(main branch) / Production

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

Add a new integration with CalDav and notice the new setup page

### CODE REVIEW

File: packages/app-store/applecalendar/pages/setup/index.tsx [hunk 1]
```diff
@@ -70,23 +70,26 @@ export default function AppleCalendarSetup() {
...
                 {errorMessage && <Alert severity="error" title={errorMessage} className="my-4" />}
-                <div className="mt-5 sm:mt-4 sm:flex sm:flex-row-reverse">
+                <div className="mt-5 space-x-2 space-x-reverse sm:mt-4 sm:flex sm:flex-row-reverse">
+                  <Button type="button" onClick={() => router.back()}>
+                    {t("cancel")}
+                  </Button>
                   <Button type="submit" loading={form.formState.isSubmitting}>
-                    Save
+                    {t("save")}
                   </Button>
```
Comment (PeerRich) [inline packages/app-store/applecalendar/pages/setup/index.tsx:93]: where does this flex-row-reverse come from all the time? lets get rid of this. likely copied from an old component.

Cancel should always be left, with `color="secondary"`  never two primary buttons

Comment (PeerRich) [reply packages/app-store/applecalendar/pages/setup/index.tsx:93]: just in case cc @Jaibles

**Review by PeerRich — CHANGES_REQUESTED** [review]

File: packages/app-store/applecalendar/pages/setup/index.tsx [hunk 2]
```diff
@@ -70,23 +70,26 @@ export default function AppleCalendarSetup() {
...
-                    label="Password"
+                    label={t("password")}
                     placeholder="•••••••••••••"
                     autoComplete="password"
                   />
                 </fieldset>
 
                 {errorMessage && <Alert severity="error" title={errorMessage} className="my-4" />}
-                <div className="mt-5 sm:mt-4 sm:flex sm:flex-row-reverse">
+                <div className="mt-5 space-x-2 space-x-reverse sm:mt-4 sm:flex sm:flex-row-reverse">
```
Comment (PeerRich) [inline packages/app-store/applecalendar/pages/setup/index.tsx:87]:
```suggestion
                <div className="mt-5 space-x-2 sm:mt-4 flex">
```

File: packages/app-store/applecalendar/pages/setup/index.tsx [hunk 3]
```diff
@@ -70,23 +70,26 @@ export default function AppleCalendarSetup() {
...
+                  <Button type="button" onClick={() => router.back()}>
```
Comment (PeerRich) [inline packages/app-store/applecalendar/pages/setup/index.tsx:88]:
```suggestion
                  <Button color="secondary" type="button" onClick={() => router.back()}>
```

File: packages/app-store/applecalendar/pages/setup/index.tsx (hunk 2 above)
Comment (PeerRich) [reply packages/app-store/applecalendar/pages/setup/index.tsx:87]: wait this is not correct.

**Comment by PeerRich** [comment]
@leog it should be like this: 
image

**Review by PeerRich — CHANGES_REQUESTED** [review]
@leog it should be like this

image

can you double check if the other set up pages have the same problem?

Cancel is always color="secondary"

**Comment by leog** [comment]
Done! Good catch @PeerRich. Here they are now:
caldav_server
apple_server

**Review by PeerRich — APPROVED** [review]

**Pull request merged**